metadata-ingestion/src/datahub/ingestion/source/sql/mssql/models.py

```diff
@@ -0,0 +1,681 @@
+
+
+class SQLServerConfig(BasicSQLAlchemyConfig):
+    # defaults
+    host_port: str = Field(default="localhost:1433", description="MSSQL host URL.")
+    scheme: str = Field(default="mssql+pytds", description="", hidden_from_schema=True)
+    include_stored_procedures: bool = Field(
+        default=True, description="Include ingest of stored procedures. Requires access to the 'sys' schema."
+    )
+    include_stored_procedures_code: bool = Field(
+        default=True, description="Include information about object code."
+    )
+    include_jobs: bool = Field(
+        default=True, description="Include ingest of MSSQL Jobs. Requires access to the 'sys' schema."
+    )
+    include_descriptions: bool = Field(
+        default=True, description="Include table descriptions information."
+    )
+    use_odbc: bool = Field(
+        default=False,
+        description="See https://docs.sqlalchemy.org/en/14/dialects/mssql.html#module-sqlalchemy.dialects.mssql.pyodbc.",
+    )
+    uri_args: Dict[str, str] = Field(
+        default={},
+        description="Arguments to URL-encode when connecting. See https://docs.microsoft.com/en-us/sql/connect/odbc/dsn-connection-string-attribute?view=sql-server-ver15.",
+    )
+    database_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for databases to filter in ingestion.",
+    )
+    database: Optional[str] = Field(
+        default=None,
+        description="database (catalog). If set to Null, all databases will be considered for ingestion.",
+    )
+
+    database_alias: Optional[str] = Field(
+        default=None,
+        description="Alias to apply to database when ingesting. Ignored when `database` is not set.",
+    )
+
+    @pydantic.validator("uri_args")
+    def passwords_match(cls, v, values, **kwargs):
+        if values["use_odbc"] and "driver" not in v:
+            raise ValueError("uri_args must contain a 'driver' option")
+        elif not values["use_odbc"] and v:
+            raise ValueError("uri_args is not supported when ODBC is disabled")
+        return v
+
+    def get_sql_alchemy_url(
+        self,
+        uri_opts: Optional[Dict[str, Any]] = None,
+        current_db: Optional[str] = None,
+    ) -> str:
+        if self.use_odbc:
+            # Ensure that the import is available.
+            import pyodbc  # noqa: F401
+
+            self.scheme = "mssql+pyodbc"
+
+        uri: str = self.sqlalchemy_uri or make_sqlalchemy_uri(
+            self.scheme,  # type: ignore
+            self.username,
+            self.password.get_secret_value() if self.password else None,
+            self.host_port,  # type: ignore
+            current_db if current_db else self.database,
+            uri_opts=uri_opts,
+        )
+        if self.use_odbc:
+            uri = f"{uri}?{urllib.parse.urlencode(self.uri_args)}"
+        return uri
+
+    def get_identifier(self, schema: str, table: str) -> str:
```

Comment:
We can remove this. Its already present in source class.

Reply:
The function should not be removed because it differs from the code in the source.

Reply:
@RChygir I looked at code for both methods and the result is will be same. The variations in method in source class is only for the case when database is not mentioned in config explicitly, i.e. , when all databases are ingested.

Reply:
Removed